cookbook/agents/README.md

```diff
@@ -0,0 +1,531 @@
+agent = Agent(
+    model=OpenAIChat(id="gpt-4o-mini"),
+    db=PostgresDb(db_url="postgresql://...", session_table="sessions"),
+    session_id="my_session",
+    add_history_to_context=True,
+)
+
+# Custom session naming
+agent.print_response("Tell me about space")
+agent.set_session_name(session_name="Space Facts Discussion")
+
+# Auto-generate session names
+agent.set_session_name(autogenerate=True)
+
+# Retrieve session details
+session = agent.get_session(session_id=agent.session_id)
+print(session.session_data.get("session_name"))
+```
+
+**Dependency Injection** - Add external data to agent context:
+```python
+import json
+
+import httpx
+from agno.agent import Agent
+from agno.models.openai import OpenAIChat
+
+
+def get_top_hackernews_stories(num_stories: int = 5) -> str:
+    """Fetch and return the top stories from HackerNews.
+
+    Args:
+        num_stories: Number of top stories to retrieve (default: 5)
+    Returns:
+        JSON string containing story details (title, url, score, etc.)
+    """
+    # Get top stories
+    stories = [
+        {
+            k: v
+            for k, v in httpx.get(
+                f"https://hacker-news.firebaseio.com/v0/item/{id}.json"
+            )
+            .json()
+            .items()
+            if k != "kids"  # Exclude discussion threads
+        }
+        for id in httpx.get(
+            "https://hacker-news.firebaseio.com/v0/topstories.json"
+        ).json()[:num_stories]
+    ]
+    return json.dumps(stories, indent=4)
+
+
+# Create a Context-Aware Agent that can access real-time HackerNews data
+agent = Agent(
+    model=OpenAIChat(id="gpt-4o"),
+    # Each function in the dependencies is resolved when the agent is run,
+    # think of it as dependency injection for Agents
+    dependencies={"top_hackernews_stories": get_top_hackernews_stories},
+    # We can add the entire dependencies dictionary to the user message
+    add_dependencies_to_context=True,
+    markdown=True,
+)
+
+# Example usage
+agent.print_response(
+    "Summarize the top stories on HackerNews and identify any interesting trends.",
+    stream=True,
+)
+
+```
+
+**Examples:**
+- **[`db/`](./db/)** - Database integration patterns
+  - `chat_history.py` - Retrieving conversation history
+  - `session_storage.py` - Persistent session storage
+  - `user_memories.py` - User-specific memory storage
+- **[`session/`](./session/)** - Advanced session management
+  - `01_persistent_session.py` - Basic session persistence
+  - `06_rename_session.py` - Custom session naming
+  - `07_in_memory_session_caching.py` - Performance optimization
+- **[`dependencies/`](./dependencies/)** - Dependency injection patterns
+  - `add_dependencies_to_context.py` - Injecting external data into agent context
+
+For more advanced patterns and enterprise features, explore:
+- **[Agno Teams](/cookbook/teams/)** - Multi-agent collaboration
+- **[Agno Workflows](/cookbook/workflows/)** - Complex orchestration
+
+---
```

Comment:
Wondering if we should close these nested READMEs with a link to the docs

Reply:
can do, but we dont have docs yet. but yes. lets keep it seperate, and we'll update with docs once we have